core/commands/profile.go

```diff
@@ -0,0 +1,234 @@
+package commands
+
+import (
+	"archive/zip"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"runtime"
+	"runtime/pprof"
+	"strings"
+	"time"
+
+	cmds "github.com/ipfs/go-ipfs-cmds"
+	"github.com/ipfs/go-ipfs/core/commands/e"
+)
+
+// time format that works in filenames on windows.
+var timeFormat = strings.ReplaceAll(time.RFC3339, ":", "_")
+
+type profileResult struct {
+	File string
+}
+
+const cpuProfileTimeOption = "cpu-profile-time"
+
+var sysProfileCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "Collect a performance profile for debugging.",
+		ShortDescription: `
+Collects cpu, heap, and goroutine profiles from a running go-ipfs daemon
+into a single zipfile. To aid in debugging, this command also attempts to
+include a copy of the running go-ipfs binary.
+`,
+		LongDescription: `
+Collects cpu, heap, and goroutine profiles from a running go-ipfs daemon
+into a single zipfile. To aid in debugging, this command also attempts to
+include a copy of the running go-ipfs binary.
+
```

Comment:
Maybe also include something like:

Profile's can be examined using `go tool pprof`, some tips can be found at https://github.com/ipfs/go-ipfs/blob/master/docs/debug-guide.md.